### PR TITLE
config option and small reformatting

### PR DESCRIPTION
**Pull request by popindavibe**
Return of the config option.

### CODE REVIEW

File: configsnap [hunk 1]
```diff
@@ -478,6 +481,20 @@ if not options.tag:
...
 
+# Load custom collection list
+Config = ConfigParser.ConfigParser()
+# Grab the full path before we chdir
+customcollectionfile = os.path.abspath(options.config)
+
+# If anything else than default was specified as custom config file
+# and is not present, raise an error and exit.
+if os.path.exists(customcollectionfile):
+    Config.read(customcollectionfile)
```
Comment (rhodesn) [inline configsnap]: I've just noticed that you are running the Config.read twice for the custom collection file, once here and once on line 537. The issue is that the read here ignores the file permissions check.


What do you think about doing the following:
```diff
diff --git a/configsnap b/configsnap
index eac5ace..239203f 100755
--- a/configsnap
+++ b/configsnap
@@ -481,20 +481,6 @@ if not options.tag:

 check_option_conflicts(options)

-# Load custom collection list
-Config = ConfigParser.ConfigParser()
-# Grab the full path before we chdir
-customcollectionfile = os.path.abspath(options.config)
-
-# If anything else than default was specified as custom config file
-# and is not present, raise an error and exit.
-if os.path.exists(customcollectionfile):
-    Config.read(customcollectionfile)
-elif customcollectionfile != '/etc/configsnap/additional.conf':
-    report_error("Additional config file %s not found! Quitting."
-                 % customcollectionfile)
-    sys.exit(1)
-
 tagdir = os.path.join(options.basedir, options.tag)
 workdir = create_dir(tagdir, options.overwrite_enabled)
 os.chdir(workdir)
@@ -523,6 +509,11 @@ if options.compare_only:
     compare_files()
     sys.exit(0)

+# Load custom collection list
+Config = ConfigParser.ConfigParser()
+# Grab the full path before we chdir
+customcollectionfile = os.path.abspath(options.config)
+
 if os.path.exists(customcollectionfile):
     # Check file is owned by root and not read/writable by anyone else
     st = os.stat(customcollectionfile)
@@ -536,6 +527,11 @@ if os.path.exists(customcollectionfile):
         # Only read in the config file if the above checks have passed
         Config.read(customcollectionfile)

+elif customcollectionfile != '/etc/configsnap/additional.conf':
+    report_error("Additional config file %s not found! Quitting."
+                 % customcollectionfile)
+    sys.exit(1)
+
 # Ensure each section has a valid Type
 for section in Config.sections():
     if Config.has_option(section, 'type'):
```

**Review by rhodesn — CHANGES_REQUESTED** [review]

File: configsnap (hunk 1 above)
Comment (popindavibe) [reply configsnap]: Good catch. Agreed 100%, I applied the changes, will test on RHEL 6/7 in the coming days.

**Comment by popindavibe** [comment]
I've moved the block above, as we need to have it **before** `chdir` to get `os.path.abspath` working as intended for `--config`.

Ready for review.

**Comment by popindavibe** [comment]
Added a 'blue' info line to mention default /etc/configsnap/additional.conf will be ignored if using `--config` and default file exsits.

**Review by rhodesn — APPROVED** [review]
Ran a sanity check against this and it all looks good, merging.

**Pull request merged**